src/node/filesystem/binary.node.ts

```diff
@@ -0,0 +1,122 @@
+import { readSync } from 'fs';
+import type { FileHandle } from 'fs/promises';
+import { open } from 'fs/promises';
+
+import type {
+  IBinaryData,
+  IBuffered,
+  IChunked,
+  IEncodable,
+  IInitializable,
+} from '../../interfaces';
+import { UnexpectedError } from '../../lib';
+
+export class ChunkedFile implements IChunked {
+  constructor(private binaryFile: BinaryFile) {}
+
+  public async getChunk(): Promise<string> {
+    return this.binaryFile.getNextChunk();
```

Comment:
issue: this seems really brittle. What if attempt something like `file.chunkBy(10).skip(1)`? This will effectively skip the last chunk and not the first one. In addition, since the state is stored in the original file it will also break if I attempt to chunk the same file twice.